packages/nx/src/plugins/js/utils/register.ts

```diff
@@ -35,6 +35,59 @@ export const registerTsProject = (
   };
 };
 
+export function getSwcTranspiler(
+  compilerOptions: CompilerOptions
+): (...args: unknown[]) => unknown {
+  // Some versions of @swc-node/register do not return a cleanup function, so we explicitly check for it here
+  type ISwcRegister = typeof import('@swc-node/register/register')['register'];
+  type ISwcRegisterVoid = (...args: Parameters<ISwcRegister>) => void;
+
+  // These are requires to prevent it from registering when it shouldn't
+  const register = require('@swc-node/register/register').register as
+    | ISwcRegister
+    | ISwcRegisterVoid;
+
+  const cleanupFn = register(compilerOptions);
+  const isFunction = typeof cleanupFn === 'function';
+
+  if (!isFunction) return () => {};
+
+  return cleanupFn;
```

Comment:
Here is the fix.

The rest of the PR is purely refactor to avoid variable reassignment and reduce the amount of nesting. e.g. early returns.

Reply:
This makes sense, swapping to a ternary to avoid extra variables